packages/fluentui/react-builder/src/components/Designer.tsx

```diff
@@ -437,6 +441,69 @@ export const Designer: React.FunctionComponent = () => {
     // FIXME: remove tree_lz from current URL
   }, [dispatch]);
 
+  const [pressedKeys, setPressedKeys] = React.useState([]);
+  const hotkeys = {
+    'Control+z': handleUndo,
+    'Shift+P': handleGoToParentComponent,
+    'Control+Shift+Z': handleRedo,
+    Delete: handleDeleteComponent,
+    'Shift+D': () => {
+      setMode('design');
+    },
+    'Shift+U': () => {
+      setMode('use');
+    },
+    'Shift+B': () => {
+      setMode('build');
+    },
+    'Shift+C': () => {
+      handleShowCodeChange(!state.showCode);
+    },
+    'Shift+J': () => {
+      handleShowJSONTreeChange(!showJSONTree);
+    },
+  };
+
+  const handleKeyDown = React.useCallback(
+    e => {
+      if (!pressedKeys.includes(e.key)) {
```

Comment:
I think it is much easier o rely on the event's `altKey`, `shiftKey`, `ctrlKey` etc. for these actions. Something like

```
handleKeyDown = e => {
  const code = getCode(e);

  if (code === keyboardKey.z && e.ctrlKey ) {
     // do something...
  }
};
```

You can see similar examples in the `Debug.tsx` component.